src/Init/Data/String/Extra.lean

```diff
@@ -14,4 +16,13 @@ if s.isNat then
 else
   panic! "Nat expected"
 
+/--
+  Convert a UTF-8 encoded `ByteArray` string to `String`.
+  The result is unspecified if `a` is not properly UTF-8 encoded. -/
```

Comment:
@leodemoura I'm assuming that, unlike in Rust, it is currently not `unsafe` to store invalid UTF-8 in a `String`. We might want "checked" alternatives in the future, but for now this should be sufficient.